sdk/python/kfp/_client.py

```diff
@@ -304,6 +304,25 @@ def create_experiment(self, name, description=None, namespace=None):
       IPython.display.display(IPython.display.HTML(html))
     return experiment
 
+  def get_pipeline_id(self, name, page_token='', page_size=100):
+    """Returns the pipeline id if a pipeline with the name exsists.
+    Args:
+      name: pipeline name
+      page_token: token for starting of the page.
+      page_size: size of the page.
+    Returns:
+      A response object including a list of experiments and next page token.
+    """
+    while True:
+        result = self.list_pipelines(page_size=page_size, page_token=page_token)
```

Comment:
this doesn't scale when number of pipelines is huge.
list_pipelines supports passing a filter, UI client's usage can be found at https://github.com/kubeflow/pipelines/blob/master/frontend/src/pages/GettingStarted.tsx#L126.

You may try to learn this usage.